AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/InitialSync/InitialSyncOrchestrator.swift

```diff
@@ -219,4 +225,39 @@ extension AWSInitialSyncOrchestrator {
         }
         return false
     }
+
+    private func isNetworkError(_ error: DataStoreError) -> Bool {
+        guard case let .sync(_, _, underlyingError) = error,
```

Comment:
is this check required if we are only looking for network error?

Reply:
This is required because the InitialSyncOrchestrator wraps the error from the operations in a `DataStoreError.sync` at line 120, stores in `syncErrors`, and moves onto the next operation.  Comes to think of it, i'm not really sure why it needs to be this way, since `syncErrors` is just an array of `[DataStoreError]`. But as it is currently, it needs to unwrap the `.sync` case like it's done above in `isUnauthorizedError`
```
if case .failure(let dataStoreError) = result {
                    let syncError = DataStoreError.sync(
                        "An error occurred syncing \(modelSchema.name)",
                        "",
                        dataStoreError)
                    self.syncErrors.append(syncError)
```